cleanlab/object_detection/summary.py

```diff
@@ -34,6 +43,412 @@
+    pred_area: DefaultDict[Any, list] = collections.defaultdict(list)
+    for sample in auxiliary_inputs:
+        _get_bbox_areas(sample["lab_labels"], sample["lab_bboxes"], labl_area, class_names)
+        _get_bbox_areas(sample["pred_labels"], sample["pred_bboxes"], pred_area, class_names)
+
+    return labl_area, pred_area
+
+
+def class_label_distribution(
+    labels=None,
+    predictions=None,
+    *,
+    auxiliary_inputs=None,
+    class_names: Optional[Dict[Any, Any]] = None,
+) -> Tuple[Dict[Any, float], Dict[Any, float]]:
+    """Returns the distribution of class labels associated with all annotated bounding boxes (or predicted bounding boxes) in the dataset.
+
+    This method can help you understand which classes are: rare or over/under-predicted by the model overall.
+
+    Parameters
+    ----------
+    labels:
+        Annotated boxes and class labels in the original dataset, which may contain some errors.
+        Refer to documentation for this argument in :py:func:`object_counts_per_image <cleanlab.object_detection.summary.object_counts_per_image>` for further details.
+
+    predictions:
+        Predictions output by a trained object detection model.
+        Refer to documentation for this argument in :py:func:`object_counts_per_image <cleanlab.object_detection.summary.object_counts_per_image>` for further details.
+
+    auxiliary_inputs: optional
+        Auxiliary inputs to be used in the computation of counts.
+        Refer to documentation for this argument in :py:func:`object_counts_per_image <cleanlab.object_detection.summary.object_counts_per_image>` for further details.
+
+    class_names: optional
+        Optional dictionary mapping one-hot-encoded class labels back to their original class names in the format ``{"integer-label": "original-class-name"}``.
+
+    Returns
+    -------
+    class_distribution: Tuple[Dict[Any, float], Dict[Any, float]]
+        A tuple containing two dictionaries. The first is a dictionary mapping each class label to its frequency in the dataset annotations.
+        The second is a dictionary mapping each class label to its frequency in the model predictions across all images in the dataset.
+    """
+    if auxiliary_inputs is None:
+        auxiliary_inputs = _get_valid_inputs_for_compute_scores(ALPHA, labels, predictions)
+
+    labl_freq: DefaultDict[Any, int] = collections.defaultdict(int)
+    pred_freq: DefaultDict[Any, int] = collections.defaultdict(int)
+    for sample in auxiliary_inputs:
+        _get_class_instances(sample["lab_labels"], labl_freq, class_names)
+        _get_class_instances(sample["pred_labels"], pred_freq, class_names)
+
+    labl_total, pred_total = sum(labl_freq.values()), sum(pred_freq.values())
+    return (
+        {k: round(v / labl_total, 2) for k, v in labl_freq.items()},
+        {k: round(v / pred_total, 2) for k, v in pred_freq.items()},
+    )
+
+
+def class_accuracy(
```

Comment:
I would call this `get_class_metrics()` as it does more than calculate class accuracy. Great work on this function!